app/src/org/commcare/tasks/ZipTask.java

```diff
@@ -128,9 +135,38 @@ private long dumpInstance(File folder, SecretKeySpec key) throws FileNotFoundExc
                 }
             }
         }
+
+        writeProperties(myDir);
+
         return FormUploadUtil.FULL_SUCCESS;
     }
 
+    private void writeProperties(File file) {
+        FileOutputStream outputStream = null;
+        try {
+            File formProperties = new File(file, "form.properties");
+            outputStream = new FileOutputStream(formProperties);
+            Properties properties = new Properties();
+            SharedPreferences settings = CommCareApplication._().getCurrentApp().getAppPreferences();
+            // HQ likes us to submit forms to the "correct" app and user specific URL
+            String postUrl = settings.getString(CommCarePreferences.PREFS_SUBMISSION_URL_KEY,
+                    c.getString(R.string.PostURL));
+            properties.setProperty("PostURL", postUrl);
+            properties.store(outputStream, null);
+        } catch(IOException e){
+            // we'll just ignore this, not the end of the world
+            e.printStackTrace();

```

Comment:
I don't really think this is a great plan. There's no reason we shouldn't fail the process entirely if this doesn't work.